src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/RangedResourceHandler.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.wrappers;
+
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+import java.util.function.Supplier;
+import net.neoforged.neoforge.transfer.TransferPreconditions;
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+
+/**
+ * A resource handler that wraps a range of indices of another handler.
+ */
+public class RangedResourceHandler<T extends IResource> extends DelegatingResourceHandler<T> {
+    protected int start;
+    protected int end;
+
+    public RangedResourceHandler(IResourceHandler<T> delegate, int start, int end) {
+        this(() -> delegate, start, end);
+    }
+
+    public RangedResourceHandler(Supplier<IResourceHandler<T>> delegate, int start, int end) {
+        super(delegate);
+        Preconditions.checkArgument(end > start, "Max index must be greater than min index");
+        this.start = start;
+        this.end = end;
+    }
+
+    public static <T extends IResource> RangedResourceHandler<T> ofSingleIndex(IResourceHandler<T> delegate, int index) {
```

Comment:
I don't like this mixing of different styles of construction for this class.

Either pick 4 different `Of(...)` or 4 Constructors.....